rfc/rfc-0018-workload-report-artifact-tree.md

```diff
@@ -0,0 +1,142 @@
+        passed:
+          - # name of the resource in the supply chain
+            resource-name: <:string>
+            # GVK of the resource
+            kind: <:string>
+            apiVersion: <:string>
+            # name of the resource on the cluster
+            name: <:string>
+            # namespace of the resource
+            namespace: <:string>
+            # resource version of the object
+            resourceVersion: <:string>
+        from:
+          - # id of the previous resource that was transformed into this one
+            id: <SHA:string>
+```
+
+example:
+Assume a supply chain that stamps out a GitRepository, Runnable, Image and config in a configmap.
+Assume that a new commit was just made, so that the GitRepository has just changed url/revision
+but the Runnable has not yet updated.
+
+Previous Source output:
+uri: https://www.some-site.com/my-project/my-repo
+revision: b974272e27c47a01e7a7da07cf8e4415bdb83dae
+
+Current Source output:
+uri: https://www.some-site.com/my-project/my-repo
+revision: b31d09004503e52e84ff633e547f4d5b40503ab3
+
+```yaml
+status:
+  artifacts:
+    - source:
+        id: e2212e77caf0cb64a25dfa1aca39599b69d72015dbb4ed2ad740f0666af35968
+        uri: https://www.some-site.com/my-project/my-repo
+        revision: b31d09004503e52e84ff633e547f4d5b40503ab3
+        passed:
+          - resource-name: source-provider
+            kind: GitRepository
+            apiVersion: source.toolkit.fluxcd.io/v1beta1
+            name: my-app
+            namespace: my-namespace
+            resourceVersion: "11125094"
+    - source:
+        id: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
+        uri: https://www.some-site.com/my-project/my-repo
+        revision: b974272e27c47a01e7a7da07cf8e4415bdb83dae
+        passed:
+          - resource-name: source-provider
+            kind: GitRepository
+            apiVersion: source.toolkit.fluxcd.io/v1beta1
+            name: my-app
+            namespace: my-namespace
+            resourceVersion: "11125090"
+          - resource-name: source-tester
+            kind: Runnable
+            apiVersion: carto.run/v1alpha1
+            name: my-app
+            namespace: my-namespace
+            resourceVersion: "11125545"
+    - image:
+        id: 373c0dc7d3cccd8ef31cd3dcf0f07b6bcc3a9ad1270db8fe43f84e26595af32a
+        image: 10.138.0.2:5000/example-testing-sc-testing-sc@sha256:9aca70a5408b7d5615724bcb8e5eea3bf0765f95eac177433993cf6002311d9b
+        passed:
+          - resource-name: image-builder
+            kind: Image
+            apiVersion: kpack.io/v1alpha2
+            name: my-app
+            namespace: my-namespace
+            resourceVersion: "11126348"
+        from:
+          - id: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
+    - config:
+        id: 22fe69c924b55327ce3e8ca079d7295a947a3641b5ab74bdf7541dc680258c81
+        config: |
```

Comment:
should this example be a map[string]string instead of a string?

Reply:
The config artifact has one field: `config` which is a string. Similar to the image artifact.